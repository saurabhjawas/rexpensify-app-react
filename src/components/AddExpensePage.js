import React from 'react'
import { connect } from 'react-redux'

import { addExpense } from '../actions/expenses'

import ExpenseForm from './ExpenseForm'

class AddExpensePage extends React.Component {
  onSubmit = (expense) => {
    // props.dispatch(addExpense(expense))
    this.props.addExpense(expense)
    this.props.history.push('/')
  }

  render() {
    return (
      <div>
        <h1>Add Expense</h1>
        <ExpenseForm
          onSubmit={this.onSubmit}
        />
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  addExpense: (expense) => {
    dispatch(addExpense(expense))
  }
})

export { AddExpensePage as AddExpensePageForTest }

export default connect(undefined, mapDispatchToProps)(AddExpensePage)