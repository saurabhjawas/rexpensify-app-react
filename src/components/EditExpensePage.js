import React from 'react'
import { connect } from 'react-redux'

import ExpenseForm from './ExpenseForm'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'

class EditExpensePage extends React.Component {
  onClick = (e) => {
    this.props.startRemoveExpense({ id: this.props.match.params.id })
    this.props.history.push('/')
  }

  onSubmit = (expense) => {

    this.props.startEditExpense({
      id: this.props.expense.id,
      updates: expense
    })

    this.props.history.push('/')
  }

  render() {
    return (
      <div>
        Editting the expense with id of {this.props.match.params.id}
        <ExpenseForm
          expense={this.props.expense}  
          onSubmit={this.onSubmit}
        />
  
        <button
          onClick={this.onClick}
        >
          Remove
        </button>
      </div>
    )
  }
}

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find(({ id }) => (id === props.match.params.id))
})

const mapDispatchToProps = (dispatch, props) => ({
  startRemoveExpense: ({id}) => {
    dispatch(startRemoveExpense({id}))
  },
  startEditExpense: ({id, updates}) => {
    dispatch(startEditExpense({ id, updates }))
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)

export { EditExpensePage as EditExpensePageForTest }

/*
const EditExpensePage = (props) => {
  // console.log(props)
  return (
    <div>
      Editting the expense with id of {props.match.params.id}
      <ExpenseForm
        expense={props.expense}

        onSubmit={(expense) => {
          // console.log('got the expense updates..', props.expense.id, expense)
          props.dispatch(editExpense({
            id: props.expense.id,
            updates: expense
          }))

          props.history.push('/')
        }}

      <button
        onClick={(e) => {
          props.dispatch(removeExpense({ id: props.match.params.id }))
          props.history.push('/')
        }}
      >
        Remove
      </button>
    </div>
  )
}
*/