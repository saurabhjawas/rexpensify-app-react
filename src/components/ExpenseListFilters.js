import React from 'react'

import { connect } from 'react-redux'

import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from '../actions/filters'

import { DateRangePicker } from 'react-dates'


class ExpenseListFilters extends React.Component {
  constructor(props) {
    super(props)
  }

  state = {
    calendarFocused: null
  }

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate)
    this.props.setEndDate(endDate)
  }

  onFocusChange = (focusedInput) => {
    this.setState(() => ({calendarFocused: focusedInput}))
  }

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value)
  }

  onSortChange = (e) => {
    switch (e.target.value) {
      case 'amount':
        this.props.sortByAmount()
        break;
    
      case 'date':
        this.props.sortByDate()
        break;

      default:
        break;
    }
  }

  render() {
    return (
      <div>
        <input 
          type="text" 
          value={this.props.filters.text} 
          onChange={this.onTextChange} 
        />
        <select
          value={this.props.filters.sortBy}
          onChange={this.onSortChange}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>

        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          numberOfMonths={1}
          isOutsideRange={() => (false)}
          showClearDates={true}
        />
    </div>
    )
  }
}

const mapStateToProps = (state) => ({
  filters: state.filters
})

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => {
    dispatch(setTextFilter(text))
  },
  sortByAmount: () => {
    dispatch(sortByAmount())
  },
  sortByDate: () => {
    dispatch(sortByDate())
  },
  setStartDate: (date) => {
    dispatch(setStartDate(date))
  },
  setEndDate: (date) => {
    dispatch(setEndDate(date))
  }
  // dispatch(setEndDate(endDate))
})

export { ExpenseListFilters as ExpenseListFiltersForTest}

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)