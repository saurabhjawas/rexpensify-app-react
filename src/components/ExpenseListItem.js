import React from 'react'
import moment from 'moment'
// import { connect } from 'react-redux'
// import { removeExpense } from '../actions/expenses'

import numeral from 'numeral'

import { Link } from 'react-router-dom'

const ExpenseListItem = ({id, description, amount, createdAt}) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>
      {numeral(amount / 100).format('$0,0.00')}
      -
      {moment(createdAt).format('DD MMMM YYYY')}</p>
  </div>
)



// export default connect()(ExpenseListItem)
export default ExpenseListItem

// {id, description, amount, createdAt, dispatch}