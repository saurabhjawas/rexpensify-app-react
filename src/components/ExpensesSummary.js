import React from 'react'

import { connect } from 'react-redux'
import getVisibleExpenses from '../selectors/expenses'
import getExpensesTotal from '../selectors/expenses-total'

import numeral from 'numeral'

const ExpensesSummary = (props) => {
  const expenseWord = props.expenseCount > 1 ? 'expenses' : 'expense'
  const formattedAmount = numeral(props.expensesTotal / 100).format('$0,0.00')
  return (
    <div>
      <h3>
        Viewing {props.expenseCount} {expenseWord} totalling {formattedAmount}
      </h3>   
    </div>
  )
}

const mapStateToProps = (state) => ({
  expenseCount: (getVisibleExpenses(state.expenses, state.filters)).length,
  expensesTotal: getExpensesTotal(getVisibleExpenses(state.expenses, state.filters))
})


export {ExpensesSummary as ExpensesSummaryForTest} 
export default connect(mapStateToProps)(ExpensesSummary)