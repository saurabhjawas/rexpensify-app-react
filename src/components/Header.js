import React from 'react'

import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { startLogOut } from '../actions/auth'


export const Header = (props) => (
   <header className="header">
      <div className="content-container">
         <div className="header__content" >
            <Link className="header__title" to="/dashboard">
               <h1>Expensify</h1>
            </Link>
            <button className="button button--logout" onClick={props.startLogOut}>Logout</button>
         </div>
      </div>
   </header>
)

const mapDispatchToProps = (dispatch) => ({
   startLogOut: () => {
      dispatch(startLogOut())
   }
})

export default connect(undefined, mapDispatchToProps)(Header)