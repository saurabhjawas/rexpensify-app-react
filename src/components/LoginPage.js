import React from 'react'
import { connect } from 'react-redux'
import { startLogin } from '../actions/auth'


const LoginPage = (props) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout_title">Expensify</h1>
      <p>It's time to get exenses under control</p>
      <button className="button" onClick={props.startLogin} >
        Login with Google
      </button>
    </div>
  </div>
)

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => {
    dispatch(startLogin())
  }
})

export { LoginPage }

export default connect(undefined, mapDispatchToProps)(LoginPage)