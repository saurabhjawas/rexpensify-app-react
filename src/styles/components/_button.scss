.button {
  background: $blue;
  border: none;
  color: white;
  display: inline-block;
  font-weight: 300;
  font-size: $font-size-large;
  padding: $s-size;
  text-decoration: none;
  line-height: 1;
}

.button--logout {
  background: none;
}

.button--secondary {
  background: #888;
}