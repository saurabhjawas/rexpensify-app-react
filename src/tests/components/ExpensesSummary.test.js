import React from 'react'

import {ExpensesSummaryForTest} from '../../components/ExpensesSummary'

import {shallow} from 'enzyme'


test('should render correctly for 1 expense', () => {
  const wrapper = shallow(
    <ExpensesSummaryForTest
      expenseCount={1}
      expensesTotal={235}
    />
  )

  expect(wrapper).toMatchSnapshot()
})

test('should render correctly for 2 expenses', () => {
  const wrapper = shallow(
    <ExpensesSummaryForTest
      expenseCount={2}
      expensesTotal={238}
    />
  )

  expect(wrapper).toMatchSnapshot()
})