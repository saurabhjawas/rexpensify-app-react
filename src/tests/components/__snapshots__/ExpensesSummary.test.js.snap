// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly for 1 expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </h3>
</div>
`;

exports[`should render correctly for 2 expenses 1`] = `
<div>
  <h3>
    Viewing 
    2
     
    expenses
     totalling 
    $2.38
  </h3>
</div>
`;
